.github/workflows/test.yml

```diff
@@ -1,48 +1,41 @@
-name: test
+name: ci
 
 on:
   push:
     branches:
-      - "main"
+      - legacy-v5
+      - main
   pull_request:
-    types: [opened, synchronize, reopened]
-
-permissions:
-  contents: read #  to fetch code (actions/checkout)
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
 
-      - name: Setup node
-        uses: actions/setup-node@v3
+      - uses: actions/setup-node@v3
```

Comment:
After setting up Node, consider:

```yml
      - name: Enable Corepack
        run: corepack enable
```

to enable Yarn 3 without the need of having legacy Yarn v1 installed.